Fix invalid Shape.TheC value

melissalinkert opened the pull request:
Noticed while testing https://github.com/glencoesoftware/ome-omero-roitool/pull/38:

- open any OMERO image in QuPath using this extension
- draw a simple shape
- send annotations back to OMERO
- try to export the annotation with ome-omero-roitool, get an exception because Shape.TheC is -1

ome-omero-roitool will get a companion PR shortly, but this should at least prevent the extension from sending an invalid value. Repeating the same steps as above with this change and a different image should successfully export the ROI to OME-XML.

melissalinkert committed:
Fix invalid Shape.TheC value

The* values should always be non-negative, -1 is invalid.

sbesson reviewed (commented):
Confirmed as part of the testing of https://github.com/glencoesoftware/ome-omero-roitool/pull/38 that the current release of the QuPath OMERO extension sets `theC` to `-1` for each Shape exported to OMERO. With this PR included, `theC` is set to `0` and the ROIs can be exported as OME-XML using the current HEAD of `ome-omero-roitool`.

Is setting this value unconditionally to the first channel the expected beahvior? An alternative default would be not to set `TheC`  which means the ROI would be applied to all channels.

melissalinkert committed:
Use Integer instead of int for `The*` attributes

int will always be serialized, but Integer will only be serialized if not null.

melissalinkert commented:
With defb154, `The*` should not be set when exporting from QuPath to OMERO. Using `Integer` instead of `int` means the values will only be serialized to JSON if not null. Since nothing in the ROI export logic changes the plane indexes, these values should always be null and thus no longer set in OMERO. ome-omero-roitool should be able to export ROIs without `The*` set.

sbesson reviewed (approved):
With the last commit, using a multi-channel multi-Z fluorescence images hosted on OMERO and exporting annotations created from QuPath, I get the following behavior

<img width="1956" alt="Screenshot 2024-05-31 at 09 35 20" src="https://github.com/glencoesoftware/qupath-extension-omero-web/assets/1355463/4e065eb0-5a0c-4241-a1f7-523ae5dc3642">
<img width="1898" alt="Screenshot 2024-05-31 at 09 35 14" src="https://github.com/glencoesoftware/qupath-extension-omero-web/assets/1355463/b2ee1d27-b659-4a09-8010-ab5a855ad58e">

All `theC`, `theZ` and `theT` properties are now unset and the ROIs will be interpreted as applying to all planes.

A possible follow-up would be to look into populating the `TheZ` value in QuPath and translate it accordingly when exporting to OMERO. My understanding is that the Z/T index was not properly populated prior to this PR so this is not a regression. It would be useful to have a second opinion from @muhanadz or anyone involved in the QuPath workflow.
